packages/core/src/html/UlNodeProcessor.ts

```diff
@@ -0,0 +1,265 @@
+import cheerio from 'cheerio';
+import get from 'lodash/get';
+import { MbNode, NodeOrText } from '../utils/node';
+
+const { processIconString } = require('../lib/markdown-it/plugins/markdown-it-icons');
+const emojiDictionary = require('../lib/markdown-it/patches/markdown-it-emoji-fixed');
+
+interface EmojiData {
+  [key: string]: string;
+}
+
+const emojiData = emojiDictionary as unknown as EmojiData;
+
+const ICON_ATTRIBUTES = ['icon', 'i-width', 'i-height', 'i-size', 'i-class'];
+
+interface IconAttributes {
+  icon?: string;
+  className?: string;
+  size?: string;
+  width?: string;
+  height?: string;
+}
+
+function classifyIcon(icon: string) {
+  const isEmoji = Object.prototype.hasOwnProperty.call(emojiData, icon);
+  const localFileRegex = new RegExp([
+    '^(./)?',
+    '[a-zA-Z0-9-._~:/?#[@!$&\'()*+,;=%]+',
+    '.(jpg|png|gif|bmp|svg|jpeg|webp|avif)$',
+  ].join(''));
+
+  const urlRegex = /^(http(s)?|ftp):\/\/[^\s/$.?#].[^\s]*$/;
+  const isImage = !isEmoji && (localFileRegex.test(icon) || urlRegex.test(icon));
+  return {
+    isEmoji,
+    isImage,
+    unicodeEmoji: isEmoji
+      ? emojiData[icon]
+      : undefined,
+  };
+}
+
+function createIconSpan(iconAttrs: IconAttributes): cheerio.Cheerio {
+  const {
+    isEmoji,
+    isImage,
+    unicodeEmoji,
+  } = classifyIcon(iconAttrs.icon!);
+
+  let spanContent;
+
+  if (isEmoji) {
+    spanContent = `<span aria-hidden="true">${unicodeEmoji}</span>`;
+  } else if (isImage) {
+    const img = cheerio(`<img src='${iconAttrs.icon}' alt='Icon'>`)
+      .css({ width: iconAttrs.width, height: iconAttrs.height, display: 'inline-block' })
+      .addClass(iconAttrs.className || '');
+    img.append('\u200B');
+
+    spanContent = cheerio('<span></span>').append(img).css({
+      'padding-bottom': '0.3em',
+      'padding-top': '0.3em',
+    });
+  } else {
+    spanContent = processIconString(iconAttrs.icon);
+  }
+
+  let spanNode = cheerio(spanContent).css({ 'font-size': 'unset', 'min-width': '16px' });
+
+  if (!isImage) {
+    spanNode = spanNode.css({ 'font-size': iconAttrs.size }).addClass(iconAttrs.className || '');
+  }
+  // Add invisible character to avoid the element from being empty
+  spanNode.append('\u200B');
+  return spanNode.css({
+    'line-height': 'unset',
+    'margin-inline-end': '0.3em',
+    'flex-shrink': '0',
+    height: '100%',
+  });
+}
+
+function updateNodeStyle(node: NodeOrText) {
+  const nodeCheerio = cheerio(node);
+
+  nodeCheerio.css({
+    'list-style-type': 'none',
+    'padding-inline-start': '0px',
+  });
+}
+
+const getIconAttributes = (node: MbNode, defaultIcon?: IconAttributes): IconAttributes | null => {
+  // If node.attribs doesn't exist, return null immediately
+  if (!node.attribs || (defaultIcon?.icon === undefined && node.attribs.icon === undefined)) {
+    return null;
+  }
+
+  // Prepare an object with default properties
+  const iconAttributes: IconAttributes = {
```

Comment:
Is this necessary? It seems all properties are overridden which makes `...iconAttributes` redundant.